src/nvim/main.c

```diff
@@ -678,12 +678,14 @@ void os_exit(int r)
   } else {
     ui_flush();
     ui_call_stop();
-    ml_close_all(true);           // remove all memfiles
   }
 
   if (!event_teardown() && r == 0) {
     r = 1;  // Exit with error if main_loop did not teardown gracefully.
   }
+  if (!ui_client_channel_id) {
+    ml_close_all(true);           // remove all memfiles
```

Comment:
Is the `ui_client_channel_id` check because we never initialized memfiles for that case?